src/output-json-alert.c

```diff
@@ -169,18 +171,37 @@ static void AlertJsonSourceTarget(const Packet *p, const PacketAlert *pa,
     jb_close(js);
 }
 
-static void AlertJsonMetadata(AlertJsonOutputCtx *json_output_ctx,
-        const PacketAlert *pa, JsonBuilder *js)
+static void AlertJsonReference(const PacketAlert *pa, JsonBuilder *jb)
+{
+    if (!pa->s->references) {
+        return;
+    }
+
+    const DetectReference *kv = pa->s->references;
+    jb_open_array(jb, "references");
+    while (kv) {
+        /* Note that the key and reference sizes have been bound
+         * checked during parsing
+         */
+        const size_t size_needed = kv->key_len + kv->reference_len + 1;
+        char kv_store[size_needed];
+        snprintf(kv_store, size_needed, "%s%s", kv->key, kv->reference);
+        jb_append_string(jb, kv_store);
```

Comment:
thought: would it make sense to add a `jb_append_printf`-ish function that would take a format string? It would also perhaps allow dropping the stack buffer, although I'm not sure if that would just move to the `jb_...` func instead or can be avoided completely.

Reply:
There are 8 places where `jb_append_printf` (or `jb_set_printf`) would be useful.

I used `git grep -B 5 -e jb_set_string -e jb_append_string|grep snprintf -A 3`